src/cli/run.js

```diff
@@ -109,7 +109,7 @@ async function run( args, options ) {
 			console.error( "Error: Process exited before tests finished running" );
 
 			const currentTest = QUnit.config.current;
-			if ( currentTest && currentTest.semaphore ) {
+			if ( currentTest && currentTest.asyncPauses.size ) {
```

Comment:
```suggestion
			if ( currentTest && currentTest.asyncPauses.size > 0 ) {
```
for consistency with your other checks on this